### PR TITLE
GHCJS patches.

### PR DESCRIPTION
**Pull request by alexanderkjeldaas**
Patches from the GHCJS project.

### CODE REVIEW

File: System/Posix/Files/Common.hsc [hunk 1]
```diff
@@ -520,7 +520,7 @@ foreign import ccall unsafe "fchown"
 -- Note: calls @ftruncate@.
 setFdSize :: Fd -> FileOffset -> IO ()
 setFdSize (Fd fd) off =
-  throwErrnoIfMinus1_ "setFdSize" (c_ftruncate fd off)
+  throwErrnoIfMinus1_ "setFdSize" (c_ftruncate fd (fromIntegral off))
```
Comment (hvr) [inline System/Posix/Files/Common.hsc]: why does this impedance mismatch occur? I'm reluctant to use a non-statically checked `fromIntegral` conversion which have been a common source of fileoffset overflow bugs... :-/

File: System/Posix/IO/Common.hsc [hunk 2]
```diff
@@ -316,7 +316,7 @@ mode2Int SeekFromEnd  = (#const SEEK_END)
 -- | May throw an exception if this is an invalid descriptor.
 fdSeek :: Fd -> SeekMode -> FileOffset -> IO FileOffset
 fdSeek (Fd fd) mode off =
-  throwErrnoIfMinus1 "fdSeek" (Base.c_lseek fd off (mode2Int mode))
+  throwErrnoIfMinus1 "fdSeek" (fmap fromIntegral $ Base.c_lseek fd (fromIntegral off) (mode2Int mode))
```
Comment (hvr) [inline System/Posix/IO/Common.hsc]: same question as above

File: System/Posix/Process/Internals.hs [hunk 3]
```diff
@@ -9,7 +10,12 @@ import Foreign
 import Foreign.C
 import System.Exit
 import System.IO.Error
+
+#ifdef ghcjs_HOST_OS
```
Comment (hvr) [inline System/Posix/Process/Internals.hs]: This seems the wrong place to kludge this around; why can't ghcjs' `GHC.Conc (Signal)` be used/fixed?

File: System/Posix/Signals.hsc [hunk 4]
```diff
@@ -432,11 +432,10 @@ installHandler :: Signal
                -> Maybe SignalSet       -- ^ other signals to block
                -> IO Handler            -- ^ old handler
 
-#ifdef __PARALLEL_HASKELL__
+##if defined(__PARALLEL_HASKELL__)
```
Comment (hvr) [inline System/Posix/Signals.hsc]: why is it necessarily to delay the macro expansion?

**Comment by Bodigrim** [comment]
Closing as stalled and supposedly superseded by a more modern effort in #205.

**Pull request closed without merging**